src/main/java/de/tum/in/www1/artemis/web/rest/UserResource.java

```diff
@@ -164,6 +166,26 @@ else if (managedUserVM.getGroups().stream().anyMatch(group -> !artemisAuthentica
         return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);
     }
 
+    /**
+     * GET /users/search : search all users by login or name (result size is limited though on purpose, see below)
+     *
+     * @param loginOrName the login or name by which to search users
+     * @return the ResponseEntity with status 200 (OK) and with body all users
+     */
+    @GetMapping("/users/search")
+    @Secured({ AuthoritiesConstants.INSTRUCTOR, AuthoritiesConstants.ADMIN })
```

Comment:
what is the difference to `@PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")`?

Reply:
They are equivalent in meaning but `PreAuthorize` seems to be the preferred approach nowadays since it is more powerful than `Secured`.

> `@Secured` annotation is a legacy Spring Security 2 annotation that can be used to configure method security. It supports more than only role-based security, but does not support using Spring Expression Language (SpEL) to specify security constraints. It is recommended to use the `@PreAuthorize` annotation in new applications over this annotation.

Source: https://stackoverflow.com/a/57394049

Reply:
`UserResource` was the only resource in the code base still using the legacy `@Secured`. I refactored all usages to equivalent `PreAuthorize` annotations instead.